### ISSUE TITLE
How to work with patches?

### ISSUE THREAD

**Issue by OrenMe** [google-github-actions/release-please-action#244]
Thank you for this GH action, I love it.
I wanted to know if I can support patch release use case:
Let's say I released a version and since that I committed 5 PRs to the main branch.
I now have a release-please PR with these 5 changes pending.
Now lets say I want to release a patch version with one of the 5 commits cherry-picked, and before the pending release-please PR is merged. 
Can this be done?
I would want to let's say:
1. Open a branch from the last release tag with some fixed prefix name like `patch/ABC`
2. Cherry-pick one of the commits from the main branch
3. have release please open a PR with the changes only in `patch/ABC` where ABC is a descriptive string that changes

**Comment by bcoe** [comment]
@OrenMe my colleague @chingor13 has been working on adding support for multiple release lines, i.e., the ability to have a `3.x` and a `4.x` release at the same time.

Once we've released this functionality to all the releasers, which is a work in progress, I think you could potentially build a similar workflow to what you're describing using iot.

**Comment by OrenMe** [comment]
Sounds amazing.
Do you have some design or active PR I can look at or just some estimate or when it might be available?

**Comment by OrenMe** [comment]
Hi @bcoe following up on our discussion in the docs PR where you mentioned https://github.com/googleapis/release-please/issues/808
Is there any progress on that? Would love to check it out if possible.

**Comment by bcoe** [comment]
@OrenMe manifest based releasers work now:

https://github.com/googleapis/release-please/blob/master/docs/manifest-releaser.md

You can see an example of them in action [here](https://github.com/googleapis/google-api-nodejs-client/pull/2582). Would be excited to find out if it works well for your use case.

**Comment by OrenMe** [comment]
hi @bcoe, I tried to read through the docs and sample but it seems like this is meant for monorepo more then to patch releases.
Am I'm missing anything here?

**Comment by bcoe** [comment]
@OrenMe you should be able to use the manifest releaser for a non-mono-repo, it has the special key "." for releasing the top level module.

At which point the manifest can be different on a branch by branch basis, I believe eliminating the issue with collisions between branches (when there are similar versions on both).

**Comment by OrenMe** [comment]
Hi @bcoe, coming back to this issue after a long while.
I still do not see how manifest based releaser solves the issue with managing release branches.
If I understood correctly it handles mono repos or different release paths.
My request was a bit different cause I want to manage release branches. I think there is another issue talking about something a long the lines of my request in #387 

Basically my tam works like this:
1. merge to main during sprint until code freeze
2. open release-* branch on code feature freeze and start committing fixes to this branch
3. when we want to release we do so from release-* branch

So for this for example I would want to have to have a wildcard option for branches to have it catch release-* branches and open dedicated PR for this branch only

Another flow is that during the development phase we need to release a patch, and main has since progressed, so we will need to:
1. open a patch-* branch from the tag we need to patch
2. work on it
3. release from that branch a tag

So for this for example I would want to have to have a wildcard option for branches to have it catch patch-* branches and open dedicated PR for this branch only

So basically what I thought is that having the default branch be able to only be one is limiting and we can have it be determined automatically by branch pattern so if I already have main branch and it has commits and an open release-please PR against it then if I open a release-* branch then the default branch for it will be the specific release-* branch and it will open a new PR for it and won't collide with the PR to main

**Comment by OrenMe** [comment]
Hi, any update on this? would love to know if this is possible or not

**Comment by bcoe** [comment]
Release please can be configured to support any numbe of branches, and you can have multiple release lines happening at the same time from these branches, e.g., `v16.x.x` for backporting fixes, and `v17.x.x` as the mainline.

Here's an example of how this is done with [yargs](https://github.com/yargs/yargs/tree/16.x.x).

To configure these complex flows, rather than release-please understanding these complex requirements, e.g., a regex on branch name, you can apply logic when cutting branches to configure release pleasea appropriately for the branch.

This is exactly what we do with our [release-brancher](https://github.com/googleapis/repo-automation-bots/tree/main/packages/release-brancher) tool.

**Comment by OrenMe** [comment]
Thanks @bcoe !
So I see what you say, this is out of scope for release-please to handle this specifically and this can be accomplished by managing the target branch manually when opening a new release branch for LTS.
I'll check if we can utilise the release-brancher for our flow, but I see this can be handled quite easily by just applying this manually when cutting a new release branch.

As a feedback I would still wanna raise that if this is such a common practice I would still recommend adding it to the tool/action.
Basically adding a release branch naming convention that can be customized and work with it that upon opening a branch with such a name the tool will automatically be able to work with it.
If I think about it no one will open by accident a branch called release-1.x or release-2.x if this is the convention of a repo for LTS branches so it is safe to assume that this can be defined by the action options.

In any case, I'll be closing the issue now as I know how to proceed.

Thank you for all the support, I love this tool!
